### PR TITLE
[13.0][MIG] purchase_order_approved

### PR DESCRIPTION
**Pull request by kongrattapong**
Standard migration

### CODE REVIEW

File: purchase_order_approved/models/purchase_order.py [hunk 1]
```diff
@@ -28,14 +28,16 @@ class PurchaseOrder(models.Model):
...
+            super(PurchaseOrder, rec).button_approve()
 
-    @api.multi
     def button_approve(self, force=False):
-        approve_purchases = self.filtered(
-            lambda p: p.company_id.purchase_approve_active)
-        approve_purchases.write({'state': 'approved'})
-        return super(PurchaseOrder, self - approve_purchases).button_approve(
-            force=force)
+        for rec in self:
```
Comment (HviorForgeFlow) [inline purchase_order_approved/models/purchase_order.py]: It should have been modified. It is not necessary to go for records as you can write on the hole recordset at once and then run button approve function that also allows multi write.

File: purchase_order_approved/models/purchase_order.py [hunk 2]
```diff
@@ -28,14 +28,16 @@ class PurchaseOrder(models.Model):
     company_id = fields.Many2one(states=READONLY_STATES)
     picking_type_id = fields.Many2one(states=READONLY_STATES)
 
-    @api.multi
     def button_release(self):
-        super(PurchaseOrder, self).button_approve()
+        for rec in self:
```
Comment (HviorForgeFlow) [inline purchase_order_approved/models/purchase_order.py]: Same here, remove for rec in self.

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]
Please split linting commit from migration one.

File: purchase_order_approved/models/purchase_order.py (hunk 1 above)
Comment (kongrattapong) [reply purchase_order_approved/models/purchase_order.py]: I see, Thank for reviews

Comment (kongrattapong) [reply purchase_order_approved/models/purchase_order.py]: @HviorForgeFlow I see, Thank for reviews

**Comment by HviorForgeFlow** [comment]
You are mixing black and migration here. https://github.com/OCA/purchase-workflow/pull/829/commits/46aba99c860f4f39a37383ed9bb6885ad02e53c0

Two commits minimum:
[IMP] purchase_order_approved:  black, isort        # precommit only
[MIG] Migration to v13: purchase_order_approved    # Migration (and modifications)
....

For the technical and functional side is ok, It is just a matter of commit history neatness.

**Comment by kongrattapong** [comment]
> You are mixing black and migration here. [46aba99](https://github.com/OCA/purchase-workflow/commit/46aba99c860f4f39a37383ed9bb6885ad02e53c0)
> 
> Two commits minimum:
> [IMP] purchase_order_approved: black, isort # precommit only
> [MIG] Migration to v13: purchase_order_approved # Migration (and modifications)
> ....
> 
> For the technical and functional side is ok, It is just a matter of commit history neatness.

@HviorForgeFlow Thank for review, commit history fixed.

**Comment by kongrattapong** [comment]
> You are mixing black and migration here. [46aba99](https://github.com/OCA/purchase-workflow/commit/46aba99c860f4f39a37383ed9bb6885ad02e53c0)
> 
> Two commits minimum:
> [IMP] purchase_order_approved: black, isort # precommit only
> [MIG] Migration to v13: purchase_order_approved # Migration (and modifications)
> ....
> 
> For the technical and functional side is ok, It is just a matter of commit history neatness.

@HviorForgeFlow I fix Commit history already. thank for review

**Comment by HviorForgeFlow** [comment]
@kongrattapong I miss some asserts in testing side.

**Comment by kittiu** [comment]
@HviorForgeFlow should this be ok, I want to proceed with another one https://github.com/OCA/purchase-workflow/pull/942
Thanks!

File: purchase_order_approved/models/purchase_order.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    state = fields.Selection(selection_add=[("approved", "Approved")])
```
Comment (HviorForgeFlow) [inline purchase_order_approved/models/purchase_order.py:10]: I think you can move the state to the proper position using this, needs to be tested.
```suggestion
    state = fields.Selection(selection_add=[("to_approve",), ("approved", "Approved")])
```

**Review by HviorForgeFlow — APPROVED** [review]
Code + functional review LGTM :+1:

**Comment by kittiu** [comment]
Code and Functional Reviewed.
Some cleanup, history and remove the README as it seem to not updated.

**Review by kittiu — APPROVED** [review]

**Review by JordiBForgeFlow — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM

**Comment by JordiBForgeFlow** [comment]
Thanks @kongrattapong and @kittiu !

**Review by AdriaGForgeFlow — APPROVED** [review]
Code review :+1:  no major changes

**Pull request merged**